cmd/geth/consolecmd_cg_test.go

```diff
@@ -97,11 +101,23 @@ func TestGethFailureToLaunch(t *testing.T) {
 	}
 }
 
+// randomStr is used in naming the geth tests' temporary datadir.
+func randomStr(n int) string {
+	letterBytes := "abcdefghijklmnopqrstuvwxyz"
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = letterBytes[rand.Int63()%int64(len(letterBytes))]
+	}
+	return string(b)
+}
+
 // TestGethStartupLogs tests that geth logs certain things (given some set of flags).
 // In these cases, geth is run with a console command to print its name (and tests that it does).
 func TestGethStartupLogs(t *testing.T) {
 	// semiPersistentDatadir is used to house an adhoc datadir for co-dependent geth test cases.
-	semiPersistentDatadir := filepath.Join(os.TempDir(), fmt.Sprintf("geth-startup-logs-test-%d", time.Now().Unix()))
+	// WATCHOUT: For Unix-based operating systems, you're going to have problems if the IPC endpoint is
+	// longer than ___ characters.
+	semiPersistentDatadir := filepath.Join(os.TempDir(), fmt.Sprintf("geth-test-%x", randomStr(4)))
```

Comment:
:speech_balloon:  @ziogaschr Looks familiar... :wink: this you?

Reply:
Yes man, copied your hack. :)